compiler/rustc_hir_analysis/src/check/intrinsicck.rs

```diff
@@ -218,8 +218,9 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
         // Check the type against the list of types supported by the selected
         // register class.
         let asm_arch = self.tcx.sess.asm_arch.unwrap();
+        let allow_experimental_reg = self.tcx.features().asm_experimental_reg();
         let reg_class = reg.reg_class();
-        let supported_tys = reg_class.supported_types(asm_arch);
+        let supported_tys = reg_class.supported_types(asm_arch, allow_experimental_reg);
         let Some((_, feature)) = supported_tys.iter().find(|&&(t, _)| t == asm_ty) else {
```

Comment:
This also needs to handle the case where a given type is only supported with allow_experimental_reg and report an appropriate error message. This would be used in the future for SVE support for example since it adds more types to an existing register class.

Reply:
Done.
https://github.com/rust-lang/rust/blob/c024d8ccdfb462f4e99b54e6c8456fbc4d745752/tests/ui/feature-gates/feature-gate-asm_experimental_reg.stderr#L11-L19